josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -34,12 +34,15 @@ type FetchTimers = HashMap<String, std::time::Instant>;
 type Polls =
     Arc<std::sync::Mutex<std::collections::HashSet<(String, josh_proxy::auth::Handle, String)>>>;
 
+type HeadsMap = Arc<std::sync::RwLock<std::collections::HashMap<String, String>>>;
```

Comment:
Why a RwLock here, or more precisely why doesn't it also benefit the places where we use Mutex?

Reply:
You are right that a Mutex would probably work just as well given how small the critical section actually is. It is a very read heavy thing though. In the end I don't think we will be able to tell the difference with regards to performance. If any of this shared structures will ever turn out be an actual bottleneck we will need something else anyway...